extensions/amp-bind/amp-bind.md

```diff
@@ -164,38 +182,105 @@ This state can be initialized with the `amp-state` component:
 </amp-state>
 ```
 
-- [Expressions](#expressions) can reference state variables via dot syntax. In this example, `myState.foo` will evaluate to `"bar"`.
-- An `<amp-state>` element's child JSON has a maximum size of 100KB.
-- `<amp-state>` can also specify a CORS URL instead of a child JSON script. See the [Appendix]((#amp-state) for details.
+[Expressions](#expressions) can reference state variables via dot syntax. In this example, `myState.foo` will evaluate to `"bar"`.
+
+Note that an `<amp-state>` element's child JSON has a maximum size of 100KB. 
 
-#### `AMP.setState()`
+`<amp-state>` can also specify a CORS URL instead of a child JSON script. See the [Appendix](#amp-state-specification) for details.
 
-State can be mutated by the [`AMP.setState()` action](../../spec/amp-actions-and-events.md).
+#### Updating state with `AMP.setState()`
 
-- `AMP.setState()` performs a deep merge of its arguments with the document state up to a depth of 10.
-- `AMP.setState()` can override variables initialized by `amp-state`.
+The [`AMP.setState()` action](../../spec/amp-actions-and-events.md) merges an object literal into the state. For example:
+
+```html
+<!-- Like JavaScript, you can reference existing
+variables in the values of the state update object literal. -->
+<button on="tap:AMP.setState({foo: 'bar', baz: myAmpState.someVariable})"></button>
+```
+
+When this button is pressed, `AMP.setState()` will [deep-merge](#deep-merge-with-ampsetstate) the update with the state up to a depth of 10. All variables, including those introduced by `amp-state`, can be overidden.
```

Comment:
Nit: "AMP.setState() will deep-merge nested objects up to a maximum depth of 10."